### DISCUSSION TITLE
It is possible to configure flexible message retry delay ?

### DISCUSSION THREAD

**Discussion by radoslawtwardy** [confluentinc/parallel-consumer#79] [Ideas] [answered]
Hi, I see method `defaultMessageRetryDelay` on `ParallelConsumerOptions`, but it only can set some fix message retry delay ? It is possible some way to set something more flexible, f. ex. exponential algorithm. I don't want retry with same duration in case of some technical problems occur (like unavailable service which I call) which take some time). 
Thanks.

**Comment by astubbs** [comment, marked as answer]
Hi @radoslawtwardy ! Welcome to the project! 

Not currently, no. However it's been planned, just waiting for someone to ask for it :)

Have you seen https://github.com/confluentinc/parallel-consumer/issues/65 ? Let's discuss it there. Let me know what sort of thing would satisfy your use case - how about providing a user function that returns a delay amount? That would be the most flexible.
